### PR TITLE
feat: add Discord integration tool

### LINKED ISSUE #1330: Add Discord as a Tool
- [How to contribute](https://github.com/phidatahq/phidata/blob/main/CONTRIBUTING.md)
- [Discord Doc](https://discordpy.readthedocs.io/en/stable/)

### PR DESCRIPTION
**Pull request by vishkrish200**
Add Discord bot functionality to phi tools with async support. Includes basic channel and message management capabilities.

- Send/delete messages
- Get channel history
- List channels and info
- Example usage in cookbook

this fixes #1330

### CODE REVIEW

**Comment by vishkrish200** [comment]
had some issues with the validate script. had to deal with the async nature of the discord.py library

File: phi/tools/discord_tools.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        Args:
+            bot_token: Discord bot token for authentication
+            enable_messaging: Whether to enable message sending functionality
+            enable_history: Whether to enable message history retrieval
+            enable_channel_management: Whether to enable channel management
+        """
+        super().__init__(name="discord_tools")
+
+        self.bot_token = bot_token
```
Comment (anuragts) [inline phi/tools/discord_tools.py:35]: Can you also add get the bot_token from env. 
example - getenv("EXA_API_KEY")
Also add logger.error if bot token is not found.

File: cookbook/tools/discord_tools.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+discord_tools = DiscordTools(
+    bot_token="your-bot-token",  # Replace with your Discord bot token
+    enable_messaging=True,
+    enable_history=True,
+    enable_channel_management=True,
+)
+
+# Create an agent with the discord toolkit
+agent = Agent(
+    instructions=[
```
Comment (anuragts) [inline cookbook/tools/discord_tools.py:19]: Add a unique name and pass model to the agent, also set markdown to true.
here is an example - https://github.com/phidatahq/phidata/blob/main/cookbook/agents_101/web_search.py

**Review by anuragts — DISMISSED** [review]

File: phi/tools/discord_tools.py (hunk 1 above)
Comment (vishkrish200) [reply phi/tools/discord_tools.py]: hey solved for this in the latest commit

File: cookbook/tools/discord_tools.py (hunk 2 above)
Comment (vishkrish200) [reply cookbook/tools/discord_tools.py:24]: solved for this too

**Comment by vishkrish200** [comment]
@anuragts . just did some more testing of the tool and it looks like it will need a few more changes to work. 

discord.py is inherently asynchronous - it requires a running event loop.
The discord bot needs to be properly started and connected before any operations.
Apologies for the delay

**Comment by vishkrish200** [comment]
hey the latest commits fix everything. switched the approach completely. using discord's http api instead of the discord bot system. updated cookbook with the appropriate steps. also included the changes @anuragts requested :)

**Comment by vishkrish200** [comment]
hey @anuragts, the pr says changes are requested but those changes were marked as resolved by you. could you have a look at the updated code again?

**Comment by manthanguptaa** [comment]
Thanks @vishkrish200 for this amazing PR. I have tested it out and it's working perfectly!

File: cookbook/tools/discord_tools.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+"""Example showing how to use Discord tools."""
```
Comment (manthanguptaa) [inline cookbook/tools/discord_tools.py:1]: you can remove this

**Review by manthanguptaa — APPROVED** [review]

**Review by vishkrish200 — COMMENTED** [review]
resolved the issue @manthanguptaa. removed the comment 👍

**Comment by manthanguptaa** [comment]
@vishkrish200 can you also edit the doc strings for the function to add the function definition. Example https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py

**Comment by vishkrish200** [comment]
> @vishkrish200 can you also edit the doc strings for the function to add the function definition. Example https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py

hey just pushed a commit which fixes this. Let me know if there are any other changes needed. also i think i've unnecessarily requested a pr review from @anuragts.

**Comment by vishkrish200** [comment]
@manthanguptaa any updates?

**Review by ysolanky — APPROVED** [review]

**Pull request merged**